src/SimpleSAML/Metadata/Sources/MDQ.php

```diff
@@ -217,8 +218,8 @@ private function writeToCache(string $set, string $entityId, array $data): void
 
         Logger::debug(sprintf('%s: Writing cache [%s] => [%s]', __CLASS__, $entityId, $cacheFileName));
 
-        /** @psalm-suppress TooManyArguments */
-        $this->fileSystem->appendToFile($cacheFileName, json_encode($data), true);
+        // using dumpFile instead of appendToFile here to replace any existing cache data
+        $this->fileSystem->dumpFile($cacheFileName, json_encode($data));
```

Comment:
You could add the JSON_THROW_ON_ERROR json constant here as well. If for ever reason `$data` can not be properly encoded, we throw an exception instead of stubling on with writing an empty cacheFile which will fail reading lateron.